structure/hash.go

```diff
@@ -286,6 +286,67 @@ func (t *TxStructure) iterateHash(key []byte, fn func(k []byte, v []byte) error)
 	return nil
 }
 
+// ReverseHashIterator is the reverse hash iterator.
+type ReverseHashIterator struct {
+	t      *TxStructure
+	iter   kv.Iterator
+	prefix []byte
+	done   bool
+	field  []byte
+}
+
+// Next implements the Iterator Next.
+func (i *ReverseHashIterator) Next() error {
+	err := i.iter.Next()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if !i.iter.Key().HasPrefix(i.prefix) {
+		i.done = true
+		return nil
+	}
+
+	_, field, err := i.t.decodeHashDataKey(i.iter.Key())
+	if err != nil {
+		return errors.Trace(err)
+	}
+	i.field = field
+	return nil
+}
+
+// Valid implements the Iterator Valid.
+func (i *ReverseHashIterator) Valid() bool {
+	return i.iter.Valid() && !i.done
+}
+
+// Key implements the Iterator Key.
+func (i *ReverseHashIterator) Key() []byte {
+	return i.field
+}
+
+// Value implements the Iterator Value.
+func (i *ReverseHashIterator) Value() []byte {
+	return i.iter.Value()
+}
+
+// Close Implements the Iterator Close.
+func (i *ReverseHashIterator) Close() {
+}
+
+// NewHashReverseIter creates a reverse hash iterator.
+func (t *TxStructure) NewHashReverseIter(key []byte) (*ReverseHashIterator, error) {
```

Comment:
How about  make `t *TxStructure` a parameter for a new func `NewHashReverseIter`. 
And make a new method `getHashReverseIter ` bind to `t *TxStructure`.
new a struct in other struct‘s method seems not a good design.

Reply:
Great.